docs/storage-config.md

```diff
@@ -0,0 +1,204 @@
+access the storage for any PV that is backend by a network block device (in
+particular the iSCSI and RBD k8s backends).
+
+    disks:
+      - source:
+          pvc:
+	    claimname: some-pvc-name
+
+When launching QEMU the PVC resource is accessed to get the corresponding
+PV resource. The configuration for the ISCSI / RBD backend is then extracted
+to configure the QEMU storage backend
+
+
+## Access modes
+
+There is no general requirement that any disk source be used exclusively by a
+single virtual machine. There are three permissible access modes
+
+* Write-exclusive: a single virtual machine is writing to the storage. No
+  other VM may access the storage whether for read or write.
+* Write-shared: multiple virtual machines are permitted to access the storage,
+  with concurrent writes permitted. This implies that the guest OS has some kind
+  of cluster-aware filesystem over the disk, details of which are opaque to the
+  host.
+* Read-shared: multiple virtual machines are pmitted to access the storage, but
+  all must be read-only. This allows arbitrary guest usage without needing any
+  kind of cluster-aware filesystems.
+
+
+## Copy mode
+
+The default behaviour will be to directly access the referenced network storage
+from a QEMU network driver. IOW any writes made by QEMU will be persistent
+against the master storage.
+
+It is, however, also valid to want local file / block device based storage for
+the virtual machine, whose content is initialized by the master referenced
+storage. This can be achieved by introducing the notion of a "copy mode" in the
+disk backend configuration. This would accept the following options
+
+* None: the storage is directly accessed by QEMU. This is the default.
+* Copy-on-write: an overlay is provided with copy-on-write semantics against the
+  master storage. The local overlay will only contain changes made by the guest.
+* Copy-on-read: an overlay is provided with copy-on-read semantics against the
+  master storage. The local overlay will contain both changes made by the guest,
+  and copies of any blocks read.
+* Clone: A full copy of the master storage is made to a local file. Once this is
+  created, no further use is made of the master storage.
+
+The first copy mode provides long term persistent storage, while the other modes
+provide ephemeral storage that is discarded when the virtual machine is deleted
+from the node.
+
+The host needs to have somewhere to storage the local overlays / files. This is
+achieved by configuring one or more volumes for the libvirtd POD when it is
+created, and mounting them into a defined location. Any valid PVC is usable for
+this purpose. If the PVC backing store is only accessible for a single host,
+then when performing migration, QEMU block storage migration feature will need
+to be enabled. If the PVC references a networked filesystem that can be mounted
+on multiple hosts, then VMs can migrate with no need to copy emphemeral disks.
```

Comment:
Whether the network storage driver has local caching or not is tangential to this choice. It can potentially improve the performance of any of these options.